hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleCreateHandler.java

```diff
@@ -142,17 +142,15 @@ public void handle(@NonNull final HandleContext context) throws HandleException
                     validate(provisionalSchedule, currentConsensusTime, isLongTermEnabled);
             if (validationOk(validationResult)) {
                 final List<Schedule> possibleDuplicates = scheduleStore.getByEquality(provisionalSchedule);
-                if (isPresentIn(context, possibleDuplicates, provisionalSchedule)) {
+                if (isPresentIn(context, possibleDuplicates, provisionalSchedule))
                     throw new HandleException(ResponseCodeEnum.IDENTICAL_SCHEDULE_ALREADY_CREATED);
-                }
-                if (scheduleStore.numSchedulesInState() + 1 > schedulingConfig.maxNumber()) {
+                if (scheduleStore.numSchedulesInState() + 1 > schedulingConfig.maxNumber())
                     throw new HandleException(ResponseCodeEnum.MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
-                }
                 // Need to process the child transaction again, to get the *primitive* keys possibly required
                 final ScheduleKeysResult requiredKeysResult = allKeysForTransaction(provisionalSchedule, context);
                 final Set<Key> allRequiredKeys = requiredKeysResult.remainingRequiredKeys();
                 final Set<Key> updatedSignatories = requiredKeysResult.updatedSignatories();
-                final long nextId = getNextId(context);
+                final long nextId = context.newEntityNum();
```

Comment:
context.newEntityNum() doesn't do config.numReservedSystemEntities() check as in getNextId(). Should we check?

Reply:
no, that method was temporary to work around a test issue that's no longer relevant.